sdk/identity/identity/src/credentials/managedIdentityCredential/cloudShellMsi.ts

```diff
@@ -53,14 +53,21 @@ function prepareRequestOptions(
 
 /**
  * Defines how to determine whether the Azure Cloud Shell MSI is available, and also how to retrieve a token from the Azure Cloud Shell MSI.
+ * Since Azure Managed Identities aren't available in the Azure Cloud Shell, we log a warning for users that try to access cloud shell using user assigned identity.
  */
 export const cloudShellMsi: MSI = {
-  async isAvailable(scopes): Promise<boolean> {
+  async isAvailable(scopes, _identityClient, clientId): Promise<boolean> {
     const resource = mapScopesToResource(scopes);
     if (!resource) {
       logger.info(`${msiName}: Unavailable. Multiple scopes are not supported.`);
       return false;
     }
+    if (clientId) {
+      logger.warning(
+        `${msiName}: Unavailable. Azure Managed Identities aren't available in the Azure Cloud Shell.`
```

Comment:
This log should be specifically that specifying a client ID won’t work. Client IDs are usually provided when users make user-assigned identities, so the phrase `user-assigned` might be useful. Have we checked at what other languages are doing?

Reply:
So, perhaps `ManagedIdentityCredential does not support user-assigned identities in the Cloud Shell environment`? What do you think?